CONTRIBUTING.md

```diff
@@ -543,10 +533,47 @@ Refer datalad/config.py for information on how to add these environment variable
 - *DATALAD_ALLOW_FAIL*:
   Instructs `@never_fail` decorator to allow to fail, e.g. to ease debugging.
 
-# Changelog section
+# Release(s) workflow
+
+## Branches
+
+- `master`: changes toward the next `MAJOR.MINOR.0` release (e.g., `0.13` as of time of writing).
+  It is also used for release candidates (with `rcX` suffix)
+- `maint`: bug fixes and minor enhancements for the latest released `MAJOR.MINOR.PATCH`
+ (e.g., `0.12.7` as of time of writing)
+- `MAJOR.MINOR`: generally not used, unless some bugfix release with a crticial bugfix is needed.
+
+## Workflow
+
+- upon release of `MAJOR.MINOR.0`, `maint` branch need to be fast-forwarded to that release
+- bug fixes to functionality released within the `maint` branch, should (HIGHLY RECOMMENDED) be
+  submitted against `maint ` branch
+- cherry-picking fixes from `master` into `maint` is allowed where needed
+- `master` branch accepts PRs with new functionality
+- `master` branch merges `maint` as frequently as possible/needed 
+
+## Code flow
+
+- new functionality is initially introduced outside of the `datalad.core.` namespace
+- new functionality should mature for at least a single full `MAJOR.MINOR` release until considered
+  for migration to `datalad.core.`
+  - convenience shim imports remain in-place in old locations for at least a single `MAJOR.MINOR` cycle
+  - they should be accompanied with deprecation warnings to trigger downstream code fixups
+  - our internal code should get adjusted immediately upon migration and testing should fail if was not
```

Comment:
AFAIK we don't currently have a build that fails if there are internal deprecation warnings.  Within the context of a CONTRIBUTING, I think "testing should fail" will be unclear (is it describing something the contributor needs to actively do? something they should expect will happen with the existing setup? ...).  And writing that makes me think that maybe CONTRIBUTING isn't the best place for this section.

Stepping back, I think we might want to reshape CONTRIBUTING with the goal of making it a bit more useful to first-time contributors, as they are likely the ones that sit down and read it.  I think it'd make sense to find another home for the more detailed reference material that is currently in CONTRIBUTING.

Reply:
Yes, I felt the same somewhat and thought to place it somewhere in docs. But then I also hate searching through docs for the right document and thought that the trail if this document would be appropriate. Also it is somewhat difficult to make a clear separation of what is useful for new vs old contributors. But indeed release management is probably clearly the one which could be elsewhere

Reply:
removed in aca2430ddcde1f68bc34e87ab4c7cbd1bf111b6e -- might resurrect it as a follow up PR for a separate discussion